wasm.md

```diff
@@ -689,9 +688,10 @@ When globals are declared, they must also be given a constant initialization val
     rule asGMut (      T:AValType   ) => const T
 
     syntax Defn       ::= GlobalDefn
-    syntax GlobalDefn ::= "(" "global" OptionalId TextGlobalType Instr ")"
-                        |     "global" GlobalType
- // ---------------------------------------------
+    syntax GlobalSpec ::= TextGlobalType Instr
+    syntax GlobalDefn ::= "(" "global" OptionalId GlobalSpec ")"
+                        |     "global" OptionalId GlobalType
+ // --------------------------------------------------------
```

Comment:
I agree it would be good to document this, but I'm not sure where. These are not just local idioms: the `*Spec` production is used for all the allocated entities (func/mem/tab/glob) because they have similar structure of definition, `Text*` as an indicator that something really just belongs to the text format is in use with `TextFormatIdx`, and we have lots of abstract helpers in many places. I'll put some explanation here, but make a note to reconsider it when we do a more complete overview of the semantics.

Reply:
I added some documentation here, and similar documentation to the other sections where a `*Spec` is introduced. What do you think?

Also, would you prefer if `TextGlobalType` was called `TextFormatGlobalType`?